src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -461,6 +477,8 @@ CTranslatorExprToDXL::CreateDXLNode(CExpression *pexpr,
 	// are no longer needed
 	CDXLNode *pdxlnNew = NULL;
 
+	pexpr->SetMotionInputSegmentsNumberForChildren();
```

Comment:
I think, this should be placed in function begin to cover two cases of return below.

Reply:
Same as the other

Reply:
> If you mean two cases above then I don't think it's necessary because our case works with motions only and those two work with table scan and external scan

This case not only for motions, and if you will move it up, then you need move and upper case!

Reply:
Done